src/bin/mkjail

```diff
@@ -51,6 +56,75 @@ HELP
 exit 0
 }
 
+_load_config() {
+    local config_file="$1" ; shift
+
+   if [ -r "$config_file" ]; then
+       . "$config_file"
+    else
+        echo "Unable to load configuration from $config_file." >&2
+        exit $EX_CONFIG
+    fi
+
+    # set defaults
+    : "${ZPOOL:=""}"
+    : "${JAILDATASET:="jails"}"
+    : "${JAILROOT:="/jails"}"
+    : "${SETS:="base"}"
+
+    if [ "$ZPOOL" = "" ]; then
+        echo "ZPOOL is required and must not be empty, " \
+            "check your configuration in $config_file." >&2
+        exit $EX_CONFIG
+    fi
+
+    if [ "$JAILDATASET" = "" ]; then
+        echo "JAILDATASET is required and must not be empty, " \
+            "check your configuration in $config_file." >&2
+        exit $EX_CONFIG
+    fi
+
+    if [ "$JAILROOT" = "" ]; then
+        echo "JAILROOT is required and must not be empty, " \
+            "it should be removed from your configuration in $config_file." >&2
+        exit $EX_CONFIG
+    fi
+
+    ds_mnt=$(
+        zfs get -H -o value mountpoint "$ZPOOL/$JAILDATASET" 2>/dev/null || \
+            echo "NOT_EXIST"
+    )
+    case "$ds_mnt" in
+        "NOT_EXIST")
+            echo "Jail dataset $ZPOOL/$JAILDATASET not found." >&2
+                    cat >&2 << EOM
+
+If this is your first time using mkjail you must first create a dataset for your jails:
+
+zfs create -o mountpoint="$JAILROOT" "$ZPOOL/$JAILDATASET"
```

Comment:
This asks the user to create the filesystem, but line 80 of `create.sh` will do this

Reply:
I had previously removed the first `zfs create` from `create.sh`. I've added it back and moved it into an `_init` function for anything that needs created/checked before the script starts doing anything. This should give an error sooner if the config is wrong rather than waiting until after it's downloaded the FreeBSD release.